src/goto-programs/goto_model.h

```diff
@@ -66,6 +67,68 @@ class goto_modelt
   }
 
   void unload(const irep_idt &name) { goto_functions.unload(name); }
+
+  // Implement the abstract goto model interface:
+
+  const goto_functionst &get_goto_functions() const override
+  {
+    return goto_functions;
+  }
+
+  const symbol_tablet &get_symbol_table() const override
+  {
+    return symbol_table;
+  }
+
+  const goto_functionst::goto_functiont &get_goto_function(
+    const irep_idt &id) override
+  {
+    return goto_functions.function_map.at(id);
+  }
+
+  bool can_produce_function(const irep_idt &id) const override
+  {
+    return goto_functions.function_map.count(id);
```

Comment:
I still prefer `find(id) != end` over this I'm-lazy-and-don't-want-to-type-so-much idiom.